hivemind/averaging/matchmaking.py

```diff
@@ -169,20 +171,21 @@ async def request_join_group(self, leader: Endpoint, expiration_time: DHTExpirat
           The originally specified leader can disband group and redirect us to a different leader
         """
         assert self.is_looking_for_group and self.current_leader is None
-        call: Optional[grpc.aio.UnaryStreamCall] = None
+        stream: AsyncIterator[averaging_pb2.MessageFromLeader] = None
         try:
             async with self.lock_request_join_group:
-                leader_stub = ChannelCache.get_stub(leader, averaging_pb2_grpc.DecentralizedAveragingStub, aio=True)
-                call = leader_stub.rpc_join_group(
+                leader_stub = self._servicer.get_stub(self._p2p, leader)
+
+                stream = leader_stub.rpc_join_group(
                     averaging_pb2.JoinRequest(
-                        endpoint=self.endpoint,
+                        endpoint=self.endpoint.to_base58(),
                         schema_hash=self.schema_hash,
                         expiration=expiration_time,
                         client_mode=self.client_mode,
                         gather=self.data_for_gather,
                     )
-                )
-                message = await asyncio.wait_for(call.read(), timeout=self.request_timeout)
+                ).__aiter__()
```

Comment:
please use [aiter from hivemind.utils.asyncio](https://github.com/learning-at-home/hivemind/blob/master/hivemind/utils/asyncio.py#L31-L34)

Reply:
Currently, `hivemind.utils.asyncio.aiter()` is not related to `__aiter__()` and does something different :) We should probably rename it and create the real `aiter()` (calling `__aiter__()`) instead.